src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,110 @@
+    private static final String REQUEST_BOOK_FROM_DB = "SELECT * FROM books WHERE id = ?";
+    private static final String GET_ALL_REQUEST_BOOK = "SELECT * FROM books";
+    private static final String UPDATE_REQUEST = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BOOK_REQUEST = "DELETE FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement preparedStatement =
+                     connection.prepareStatement(INSERT_REQUEST, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.executeUpdate();
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert book data to DB. Book:" + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Book book = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement getStatement = connection.prepareStatement(REQUEST_BOOK_FROM_DB);
+            getStatement.setLong(1, id);
+            ResultSet resultSet = getStatement.executeQuery();
+            while (resultSet.next()) {
+                book = getBookFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book by id:" + id, e);
+        }
+        return Optional.ofNullable(book);
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+        PreparedStatement getAllStatement = connection.prepareStatement(GET_ALL_REQUEST_BOOK)) {
+            ResultSet resultSet = getAllStatement.executeQuery();
+            if (resultSet.next()) {
+                books.add(getBookFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cant get all books from DB.", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+        PreparedStatement updateStatement = connection.prepareStatement(UPDATE_REQUEST)) {
+            updateStatement.setLong(ID_INDEX, book.getId());
+            updateStatement.setString(TITLE_INDEX, book.getTitle());
+            updateStatement.setBigDecimal(PRICE_INDEX, book.getPrice());
```

Comment:
Does it suit your query? I mean, Does the id have first place in the query and etc?